tests/pandas_utils/test_random.py

```diff
@@ -0,0 +1,263 @@
+        ),
+        np.random.default_rng(10),
+    )
+
+
+@pytest.fixture(name="int_no_lower", scope="function")
+def fixture_int_no_lower() -> DataGeneratorRun:
+    """Test params for IntGenerator without lower value passed."""
+    return DataGeneratorRun(
+        random.IntGenerator("Steves_phone_number", 11, 9),
+        pd.Series(
+            [
+                6,
+                8,
+                2,
+                1,
+                7,
+                7,
+                4,
+                1,
+                7,
+                4,
+                1,
+            ],
+            name="Steves_phone_number",
+        ),
+        np.random.default_rng(10),
+    )
+
+
+@pytest.fixture(name="int_lower", scope="function")
+def fixture_int_lower() -> DataGeneratorRun:
+    """Test params for IntGenerator with lower value passed."""
+    return DataGeneratorRun(
+        random.IntGenerator("Steves_phone_number", 11, 9, 2),
+        pd.Series(
+            [
+                7,
+                8,
+                3,
+                3,
+                7,
+                7,
+                5,
+                3,
+                7,
+                5,
+                3,
+            ],
+            name="Steves_phone_number",
+        ),
+        np.random.default_rng(10),
+    )
+
+
+@pytest.fixture(name="id_no_starting_val", scope="function")
+def fixture_id_no_starting_val() -> DataGeneratorRun:
+    """Test params for UnqiueIdGenerator without starting value passed."""
+    return DataGeneratorRun(
+        random.UniqueIdGenerator(
+            "Steves_least_favourite_numbers",
+            20,
+        ),
+        pd.Series(
+            [
+                0,
+                1,
+                2,
+                3,
+                4,
+                5,
+                6,
+                7,
+                8,
+                9,
+                10,
+                11,
+                12,
+                13,
+                14,
+                15,
+                16,
+                17,
+                18,
+                19,
+            ],
+            name="Steves_least_favourite_numbers",
+        ),
+        np.random.default_rng(10),
+    )
+
+
+@pytest.fixture(name="id_starting_val", scope="function")
+def fixture_id_starting_val() -> DataGeneratorRun:
+    """Test params for UnqiueIdGenerator with starting value passed."""
+    return DataGeneratorRun(
+        random.UniqueIdGenerator("Steves_least_favourite_numbers", 20, 4),
+        pd.Series(
+            [
+                4,
+                5,
+                6,
+                7,
+                8,
+                9,
+                10,
+                11,
+                12,
+                13,
+                14,
+                15,
+                16,
+                17,
+                18,
+                19,
+                20,
+                21,
+                22,
+                23,
+            ],
```

Comment:
When hardcoding lists like this I would turn off black formatting to make them more concise e.g.
```suggestion
            # fmt: off
            [
                4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14,
                15, 16, 17, 18, 19, 20, 21, 22, 23,
            ],
            # fmt: on
```